### PR TITLE
Download the correct tools for Windows 64-bit

### PR DESCRIPTION
**Pull request by MatteoPologruto**
**Please check if the PR fulfills these requirements**

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-create-agent/pulls)
      before creating one)
- [x] Tests for the changes have been added (for bug fixes / features)

* **What kind of change does this PR introduce?**
Bug fix

- **What is the current behavior?**
Tools for Windows 32-bit are downloaded when the OS/Arch is Windows 64-bit.

* **What is the new behavior?**
The correct tools are downloaded for Windows 64-bit.

- **Does this PR introduce a breaking change?**
No

### CODE REVIEW

File: tools/download_test.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func TestDownloadCorrectPlatform(t *testing.T) {
 		{"darwin", "amd64", "x86_64-apple-darwin"},
 		{"darwin", "arm64", "arm64-apple-darwin"},
 		{"windows", "386", "i686-mingw32"},
-		{"windows", "amd64", "i686-mingw32"},
+		{"windows", "amd64", "x86_64-mingw32"},
```
Comment (cmaglie) [inline tools/download_test.go:43]: The Windows 32 version must be downloaded if the Windows 64 is missing.

We could implement a compatibility logic similar to the CLI:
https://github.com/arduino/arduino-cli/blob/master/arduino/cores/tools.go#L129-L216

I don't know how much work is to import directly `cores.Tools` from arduino-cli (so we can implement the same logic also for macosx), probably it's not worth it...

Anyway, IMHO, at least the compatibility rule for Windows 64 -> Windows 32 must be implemented.

**Review by cmaglie — CHANGES_REQUESTED** [review]

File: tools/download_test.go
Comment (umbynos) [inline tools/download_test.go:1]: I would also add a test to see if the correct tool is downloaded with the other os/arch combo

File: tools/testdata/test_tool_index.json [hunk 2]
```diff
@@ -1,117 +1,130 @@
...
+          "systems": [
+            {
+              "host": "i686-mingw32",
+              "url": "http://downloads.arduino.cc/arduino-fwuploader/arduino-fwuploader_2.4.0_Windows_32bit.zip",
+              "archiveFileName": "arduino-fwuploader_2.4.0_Windows_32bit.zip",
+              "checksum": "SHA-256:548447ce2028eb26257d2fdaa193680c61ccea65b0de981c60c6439d2397faae",
+              "size": "9887673"
+            }
+          ]
         }
```
Comment (umbynos) [inline tools/testdata/test_tool_index.json:126]: [This](https://github.com/arduino/arduino-create-agent/pull/813/commits/2617f628f2985b1f22fd1b02593aa2f09e0d44da?diff=split&w=1#r1303003224) could be done by adding here another version of the tool with all the os/arch available and trying to download it

**Comment by umbynos** [comment]
Also I would aim to remove completely the `smetrics.Jaro(...)` approach in favour of the CLI one:
- https://github.com/MatteoPologruto/arduino-create-agent/blob/2617f628f2985b1f22fd1b02593aa2f09e0d44da/v2/pkgs/tools.go#L253
- https://github.com/MatteoPologruto/arduino-create-agent/blob/2617f628f2985b1f22fd1b02593aa2f09e0d44da/tools/tools.go#L109

**Review by umbynos — APPROVED** [review]

**Review by cmaglie — APPROVED** [review]

**Pull request merged**